mysql-test/main/ssl_7937,nossl.result

```diff
@@ -3,15 +3,13 @@ select if(variable_value > '','yes','no') as 'have_ssl'
   from information_schema.session_status
 where variable_name='ssl_cipher';
 mysql --ssl-ca=cacert.pem -e "call test.have_ssl()"
-have_ssl
-no
+ERROR 2026 (HY000): Client requires TLS/SSL, but the server does not support it
 mysql --ssl -e "call test.have_ssl()"
-have_ssl
-no
+ERROR 2026 (HY000): Client requires TLS/SSL, but the server does not support it
```

Comment:
This is changes because this PR incorporates https://github.com/mariadb-corporation/mariadb-connector-c/pull/226 as a submodule, and that includes the fix for [MDEV-28634](https://jira.mariadb.org/browse/MDEV-28634).

Without that change, the command-line client can be **instructed to use SSL** and then _**silently downgraded**_ to an insecure/plaintext connection. :scream:

With that change, Connector/C based clients with TLS enabled (e.g. `mysql --ssl`) properly refuse to connect to servers which don't support TLS.

Reply:
I haven't checked, but I suspect the submodule update doesn't work because it links to a commit ID that isn't in the main mariadb-connector-c repo.

Reply:
> I haven't checked, but I suspect the submodule update doesn't work because it links to a commit ID that isn't in the main mariadb-connector-c repo.

Hmmm… is there anything in the Buildbot log that indicates that this is an issue? I don't see anything, or maybe don't have access to the logs that would show it.

From _Github's_ view of things, everything appears correct:

1. The "Changes" tab of this PR shows the intended change in the `libmariadb` submodule:

    ![image](https://github.com/MariaDB/server/assets/128716/138919d9-c053-4e05-b40f-fe602c65c849)

2. The target commit is indeed visible in the web-based interface for the main/upstream `mariadb-connector-c` repo (e.g. at `https://github.com/mariadb-corporation/mariadb-connector-c/commit/abad64b9cf46b0f9d6b85bf8a59f513856ea75fb`), with the usual caveat or warning message about how this commit comes from a fork:

    ![image](https://github.com/MariaDB/server/assets/128716/a2d89340-31ed-4c1d-b729-06a1827f74b2)